### PR TITLE
fix(derive): Release breaking change as v0.31.0

### PR DESCRIPTION
**Pull request by mxinden**
## Description

`libp2p-swarm-derive` released a breaking change (https://github.com/libp2p/rust-libp2p/pull/3011) as a patch release `v0.30.2`.

This patch:

1. Prepares a new minor release of `libp2p-swarm-derive`.
2. Prepares a patch release for `libp2p-swarm` to use the minor release of `libp2p-swarm-derive`.

As a follow up we can yank the `libp2p-swarm-derive` `v0.30.2` release. In addition we might want to release `libp2p-swarm-derive` `v0.30.3` containing all patches but https://github.com/libp2p/rust-libp2p/pull/3011.

See https://github.com/libp2p/rust-libp2p/issues/3176 for corresponding discussion.

## Notes

//CC @joao-paulo-parity @dvdplm

## Open Questions

If I am not mistaken, we can release this in `libp2p-swarm` as a patch release. Would appreciate additional eyes on that.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/Cargo.toml [hunk 1]
```diff
@@ -46,7 +46,7 @@ libp2p-identify = { path = "../protocols/identify" }
 libp2p-kad = { path = "../protocols/kad" }
 libp2p-ping = { path = "../protocols/ping" }
 libp2p-plaintext = { path = "../transports/plaintext" }
-libp2p-swarm-derive = { version = "0.30.2", path = "../swarm-derive" }
```
Comment (mxinden) [inline swarm/Cargo.toml:49]: `version` was introduced in https://github.com/libp2p/rust-libp2p/pull/3055/files#r1034571199. @thomaseizinger was that a deliberate choice or an oversight?

Comment (thomaseizinger) [reply swarm/Cargo.toml:49]: An oversight! Thanks for fixing it!

File: swarm/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 0.41.1
+
+- Update to `swarm-derive` `v0.31.0`.
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
- Update to `libp2p-swarm-derive` `v0.31.0`.
```

File: swarm-derive/Cargo.toml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name = "libp2p-swarm-derive"
 edition = "2021"
 rust-version = "1.60.0"
 description = "Procedural macros of libp2p-core"
```
Comment (thomaseizinger) [inline swarm-derive/Cargo.toml]:
```suggestion
description = "Procedural macros of libp2p-swarm"
```

**Comment by thomaseizinger** [comment]
> If I am not mistaken, we can release this in `libp2p-swarm` as a patch release. Would appreciate additional eyes on that.

I agree.

**Review by thomaseizinger — APPROVED** [review]

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
thanks for this Max!

**Comment by shamil-gadelshin** [comment]
Thanks indeed! I also encountered the issue.

**Comment by bkchr** [comment]
Can we move on here @mxinden?

**Comment by mxinden** [comment]
I will publish and tag the release after the libp2p community call.

**Comment by mxinden** [comment]
- libp2p-swarm-derive [published](https://crates.io/crates/libp2p-swarm-derive/0.31.0) and [tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-swarm-derive-v0.31.0)
- libp2p-swarm [published](https://crates.io/crates/libp2p-swarm/0.41.1) and [tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-swarm-v0.41.1).

**Comment by bkchr** [comment]
Ty @mxinden :)

**Comment by Voxelot** [comment]
FYI might be worth checking out `cargo-semver-checks` to detect these issues in the future. Or at the very least use some kind of snapshot tests of the codegen.

**Comment by mxinden** [comment]
@Voxelot already part of our CI see https://github.com/libp2p/rust-libp2p/pull/2647.

Though in this case, I don't think it could have caught the breaking change, given that `libp2p-swarm-derive` is a derive macro.

**Pull request merged**